src/pages/promises/[...slug].js

```diff
@@ -172,6 +178,16 @@ export async function getStaticProps({ params: { slug: slugParam }, locale }) {
   const promiseStatuses = await backend.promises({ id }).statuses;
   const site = await backend.sites().current;
 
+  let articles = await backend.articles().all;
+
+  articles = articles.filter((article) =>
+    article.categories.some(
+      (category) =>
+        promisePost.categories.map((cat) => cat.name).indexOf(category.name) !==
+        -1
+    )
+  );
+
```

Comment:
Works but a better API design would be to have this in the backend as well, on articles or promises. Either:

1. modify `backend.promises({ id }).first` to include related articles automatically since they're all coming from the same backend, or
2. add a `backend.articles({ categories: <list of categories }).all` or `backend.articles().filter({ categories: <list of categories> })` that will return all articles matching filter.